pages/api/aemService.js

```diff
@@ -0,0 +1,51 @@
+export class AEMService {
+  constructor(baseUrl, cacheBust) {
+    if (!baseUrl?.trim?.()) throw new Error(`Provide a base URL for AEM.`)
+
+    this.cacheBustString = !!cacheBust?.trim?.()
+      ? cacheBust
+      : new Date().toLocaleDateString('en-CA')
+    this.baseUrl = baseUrl
+
+    // maintain all fragments by their path
+    this.store = new Map()
+  }
+
+  async getFragment(fragPath) {
+    if (!fragPath?.trim?.()) return
+
+    // return memoized fragPath data if it's stored
+    if (this.store.has(fragPath)) {
+      return { data: this.store.get(fragPath), error: null }
+    }
+
+    // otherwise, fetch from AEM
+    const res = await fetch(
+      `${this.baseUrl}${fragPath}?dates=${this.cacheBustString}`
+    )
+    const error = res.ok ? false : res.status
+    const data = res.ok ? await res.json() : null
+
+    // if there's no error, store for memoization
+    if (!error) {
+      this.store.set(fragPath, data)
+    }
+
+    return { data, error }
+  }
+
+  async getElements(elementsPath) {
+    const { data, error } = await this.getFragment(elementsPath)
+    return {
+      elements: data?.properties?.elements || [],
+      error: error,
+    }
+  }
+}
+
+const aemServiceInstance = new AEMService(
```

Comment:
I think it's fine to work in this way since the service should be terminally blocked (or broken) if the environment vars are missing. In other words, if that service is broken for any reason and even if calls fail to AEM, it should be a terminal error that breaks the build. But please, do let me know your thoughts.